vlog4j-parser/src/test/java/org/semanticweb/vlog4j/syntax/parser/RuleParserTest.java

```diff
@@ -438,4 +454,82 @@ public void testCustomDatatype() throws ParsingException {
 		DatatypeConstant result = (DatatypeConstant) literal.getConstants().toArray()[0];
 		assertEquals(constant, result);
 	}
+
+	@Test
+	public void testGetFacts() throws ParsingException, IOException {
+		KnowledgeBase kb = new KnowledgeBase();
+		final DataSource dataSource1 = new SparqlQueryResultDataSource(new URL("https://example.org/"), "var",
```

Comment:
we should not use SparqlQueryResultDataSources in unit tests, as the tests may fail if the internet connection is low.